lib/rspec/core/configuration_options.rb

```diff
@@ -9,7 +9,7 @@ module Core
     class ConfigurationOptions
       # @param args [Array<String>] command line arguments
       def initialize(args)
-        @args = args.dup
+        @args = args.delete_if { |arg| arg =~ /rspec$/ }.dup
```

Comment:
`delete_if` modifies the Array in place, so I think you need the dup beforehand so you don't modify the caller args